### ISSUE TITLE
Use cleanSource and/or cleanSourceWith to avoid cruft

### ISSUE THREAD

**Issue by lionello** [svanderburg/node2nix#214]
I find myself always editing the generated `node-packages.nix` and changing the last `src = ./.;` into:
```
    src = lib.cleanSourceWith { filter = name: type: (builtins.baseNameOf name) != "node_modules"; src = lib.cleanSource ./.; };
```
This would also take care of *"WARNING: There is a node_modules/ folder in the root directory of the project!"*

(That expression uses `lib`, which would get passed in from the composition Nix file)

**Comment by svanderburg** [comment]
Indeed, in most cases you typically don't want to pollute your deployments with the `node_modules/` folder of the project, but sometimes this can be handy -- for example, for experimentation, or when you want to work on a project that ships bundled dependencies.
